File: android/app/src/main/java/com/redscreenfilter/data/OverlaySettings.kt
package com.redscreenfilter.data

/**
 * OverlaySettings Data Model
 * Represents the complete state of overlay configuration
 */
data class OverlaySettings(
    val isEnabled: Boolean = false,
    val opacity: Float = 0.5f,
    val scheduleEnabled: Boolean = false,
    val scheduleStartTime: String = "21:00", // HH:mm format
    val scheduleEndTime: String = "07:00",
    val useAmbientLight: Boolean = false,
    val useLocationSchedule: Boolean = false,
    val colorVariant: String = "red_standard"
)

File: ios/RedScreenFilter/Sources/Models/OverlaySettings.swift
//
//  OverlaySettings.swift
//  RedScreenFilter
//
//  Created on March 5, 2026.
//

import Foundation

struct OverlaySettings: Codable {
    var isEnabled: Bool = false
    var opacity: Float = 0.5
    var scheduleEnabled: Bool = false
    var scheduleStartTime: String = "21:00"
    var scheduleEndTime: String = "07:00"
    var useAmbientLight: Bool = false
    var useLocationSchedule: Bool = false
    var sunsetOffsetMinutes: Int = 0
    var colorVariant: String = "red_standard"
    
    enum CodingKeys: String, CodingKey {
        case isEnabled
        case opacity
        case scheduleEnabled
        case scheduleStartTime
        case scheduleEndTime
        case useAmbientLight
        case useLocationSchedule
        case sunsetOffsetMinutes
        case colorVariant
    }
}
